launcher/FileSystem.cpp

```diff
@@ -872,16 +831,35 @@ QString RemoveInvalidPathChars(QString path, QChar replaceWith)
     invalidChars = BAD_WIN_CHARS;
 #endif
 
-    QString fileSystemType = getFileSystemType(QFileInfo(path).absolutePath());
-
-    if (fileSystemType == "FAT32") {
-        invalidChars += BAD_FAT32_CHARS;
-    } else if (fileSystemType == "NTFS") {
-        invalidChars += BAD_NTFS_CHARS;
-    } else if (fileSystemType == "HFS+" || fileSystemType == "APFS") {
-        invalidChars += BAD_HFS_CHARS;
-    } else if (fileSystemType == "exFAT") {
-        invalidChars += BAD_EXFAT_CHARS;
+    switch (statFS(path).fsType) {
+        case FilesystemType::FAT:
+            invalidChars += BAD_FAT_CHARS;
+            break;
+        case FilesystemType::NTFS:
+            invalidChars += BAD_NTFS_CHARS;
+            break;
+        // case FilesystemType::REFS:
+        // case FilesystemType::EXT:
+        // case FilesystemType::EXT_2_OLD:
+        // case FilesystemType::EXT_2_3_4:
+        // case FilesystemType::XFS:
+        // case FilesystemType::BTRFS:
+        // case FilesystemType::NFS:
+        // case FilesystemType::ZFS:
```

Comment:
The only invalid "path" character on these filesystesms would be a null byte `\0` (also invalid on windows), but I'm not sure that needs to be checked for. but I do feel it should be commented on in this code to avoid confusion from others